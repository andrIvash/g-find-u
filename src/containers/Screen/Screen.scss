@import '../../styles/abstracts/breakpoints';
@import '../../styles/abstracts/functions';

.screen {
    display: flex;
    min-height: 100vh;
    flex-direction: column;
    background-color: goldenrod;
    width: rem(1024);
    padding: 0 rem(20);

    @include break-min-max(rem(425), rem(1100)) {
        width: rem(425);
        padding: 0 rem(10);
    }

    @media screen {
        @include break-min-max(rem(320), rem(500)) {
            width: rem(320);
            padding: 0 rem(5);
        }
    }
}

.screen__content {
    flex: 1;
}

.screen--dark {
    background-color: rgb(227, 206, 152);
}

