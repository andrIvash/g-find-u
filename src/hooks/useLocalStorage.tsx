import * as React from 'react';

export const useLocalStorage = (storageKey: string, fallbackState: boolean) => {
  const [value, setValue] = React.useState(
    JSON.parse(localStorage.getItem(storageKey)) ?? fallbackState
  );

  React.useEffect(() => {
    localStorage.setItem(storageKey, JSON.stringify(value));
  }, [value, storageKey]);

  return [value, setValue];
};
