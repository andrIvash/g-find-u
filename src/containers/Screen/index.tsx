import { Screen } from './Screen';

export type Props = {
    children: JSX.Element
}


export default Screen;