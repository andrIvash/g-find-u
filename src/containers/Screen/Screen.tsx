import React, { useEffect } from 'react';
import { DarkThemeContext } from '../../context/darkThemeContext';
import { Props } from './';
import Header from '../../components/Header';
import Footer from '../../components/Footer';
import SimpleSwitch from '../../components/common/SimpleSwitch';
import { useLocalStorage } from '../../hooks/useLocalStorage';
import { icons } from '../../components/common/Icons';

import './Screen.scss';

export const Screen = ({children}: Props) => {
    const [darkTheme, setDarkTheme] = useLocalStorage('dark-mode', false);
    
    const toggleTheme = () => {
        setDarkTheme(!darkTheme);
    }
    
    useEffect(() => {
        if (darkTheme) {
          document.body.classList.add('dark-mode');
        } else {
          document.body.classList.remove('dark-mode');
        }
      }, [darkTheme]);

    return (
        <DarkThemeContext.Provider value={darkTheme}>
            <div className={`screen ${darkTheme ? "screen--dark" : ""}`}>
                <div className='screen__controls center '>
                    <div className='item'>
                        <img className='icon rotate' src={icons.react} /> React
                    </div>
                    <SimpleSwitch
                        checked={darkTheme}
                        onChange={toggleTheme}
                        leftLabel={<img className='icon' src={icons.sun} />}
                        rightLabel={<img className='icon' src={icons.moon} />}
                    />
                    <button onClick={toggleTheme}>
                        {darkTheme ? 'Light Theme' : 'Dark Theme'}
                    </button>
                </div>
                <Header />
                <div className='screen__content'>
                    {children}
                </div>
                <Footer />
            </div>
        </DarkThemeContext.Provider>
  );
};