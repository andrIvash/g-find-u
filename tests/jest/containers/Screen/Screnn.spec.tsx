import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Screen from '../../../../src/containers/Screen';

jest.mock('../../../../src/components/Header', () => () => <div data-testid='mock-header'>Header</div>);
jest.mock('../../../../src/components/Footer', () => () => <div data-testid='mock-footer'>Footer</div>);
jest.mock('../../../../src/components/common/SimpleSwitch', () => () => <div data-testid='mock-simple-switch'>SimpleSwitch</div>);

jest.mock('../../../../src/hooks/useLocalStorage', () => ({
  useLocalStorage: jest.fn()
}));

const mockChildren = <div>{'children'}</div>;

describe('Screen Component', () => {
    beforeEach(() => {
        require('../../../../src/hooks/useLocalStorage').useLocalStorage.mockImplementation((key, initialValue) => {
            const [state, setState] = React.useState(initialValue);
            return [state, setState];
        });
    });

    it('renders without crashing', () => {
        render(<Screen>{mockChildren}</Screen>);
        expect(screen.getByTestId('Screen')).toBeInTheDocument();
    });

    it('toggles dark mode', () => {
        const { getByTestId } = render(<Screen>{mockChildren}</Screen>);
        fireEvent.click(getByTestId('mock-simple-switch'));
    });

    it('renders children properly', () => {
        const { getByText } = render(<Screen><div>Test Child</div></Screen>);
        expect(getByText('Test Child')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Screen>{mockChildren}</Screen>);
        expect(asFragment()).toMatchSnapshot();
    });
});