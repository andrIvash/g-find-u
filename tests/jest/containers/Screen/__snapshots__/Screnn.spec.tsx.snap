// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="screen"
    data-testid="Screen"
  >
    <div
      class="screen__controls"
    >
      <div
        data-testid="mock-simple-switch"
      >
        SimpleSwitch
      </div>
    </div>
    <div
      data-testid="mock-header"
    >
      Header
    </div>
    <div
      class="screen__content"
    >
      <div>
        children
      </div>
    </div>
    <div
      data-testid="mock-footer"
    >
      Footer
    </div>
  </div>
</DocumentFragment>
`;
